### ISSUE TITLE
NVDA does not announce Copied to clipboard in windows mail

### ISSUE THREAD

**Issue by gauravahir28** [nvaccess/nvda#16375]
### Steps to reproduce:
- Open windows mail application.
- Open any of the email containing textual content.
- Select the text using shift + arrow keys.
- Press CTRL + C to copy.
### Actual behavior:

NVDA doesn't announce "Copied to clipboard, [copied content]"
### Expected behavior:

NVDA should announce "Copied to clipboard, [copied content]"
#### NVDA installed/portable/running from source:
Installed
#### NVDA version:
NVDA 2024.1
#### Windows version:
Windows 11, 23H2
#### Name and version of other software in use when reproducing the issue:
Windows mail Version 16005.14326.21854.0
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
Works with NVDA 2023.3
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes

**Comment by XLTechie** [comment]
I'm curious. You say this works in NVDA 2023.3.
Would you please try NVDA 2023.3 with add-ons disabled, and find out if it still 
works then?

Also, be aware that Windows Mail is end of life. It will be replaced by the free 
version of Outlook sometime soon.

**Comment by gauravahir28** [comment]
> I'm curious. You say this works in NVDA 2023.3.
Would you please try NVDA 2023.3 with add-ons disabled, and find out if it still
works then?

In NVDA 2023.3, it works even when add-on's are disabled!

> Also, be aware that Windows Mail is end of life. It will be replaced by the free
version of Outlook sometime soon.

Yes, I am aware of this fact, but I am not happy with the new Outlook application. Hence, I have decided to stick with Windows Mail until I found a better mail client.

**Comment by CyrilleB79** [comment]
Same result with Microsoft Outlook (Office 2016) or Word with browse mode enabled.

The UIA parameter needs to be considered here:
1. With UIA Word document access (and browse mode on):
   * NVDA 2023.3 reports the copy action but only copies raw text
   * NVDA 2024.1 does not report the copy action but copies the formatted text

   The copy of the formatted text is now probably directly performed by the application. Thus NVDA can't probably check if the copy has occurred; thus it is expected that it does not report anything in this case.

2. With legacy Word document access (and browse mode on):
   * NVDA 2023.3 reports the copy action and copies formatted text
   * NVDA 2024.1 does not report the copy action but copies the formatted text

   Why doesn't NVDA report the formatted text copy with NVDA 2024.1, whereas it was able to report it with NVDA 2023.3? It's the only case in which something unexpected happens. Cc @michaelDCurran, it's probably a side effect of the rework during native copy implementation.

**Comment by Adriani90** [comment]
Thus I am closing as invalid.

**Comment by CyrilleB79** [comment]
> @CyrilleB79 wrote:
> 
> > With legacy Word document access (and browse mode on):
> > ◦ NVDA 2023.3 reports the copy action and copies formatted text
> 
> That was actually the part that has been replaced by the native copy message afterwards, but since we decided to let the application handle the copy process of formated text, it can not be ensured that it was successful as you described above. In 2023.3 NVDA reported "copied to clipboard" which is all right for browsers because it copies the raw text. Since in MS Word browse mode NVDA 2024.1 copies formated text, the behavior both in browse and focus mode is the same. In focus mode NVDA does not report anything as well. So in short, the message is completely handled by the application whenever formated text is copied unless you use Firefox with native selection on.

@Adriani90, please re-read my last comment in detail. Maybe I've not been clear enough.

The most illustrative case is the following:
Microsoft Outlook (2016) using legacy access to Word document: both with NVDA 2023.3 and 2024.1, the text being copied is formatted. With 2023.3, the copy action was reported by NVDA; with 2024.1 the copy action is not reported anymore. Thus, this is a regression.
So I'll re-open the issue.

Note that the test with Outlook is better easier to understand since UIA notification of "Copy" action are raised in Word but not in Outlook.

**Comment by Adriani90** [comment]
I still think it is best to keep this consistency further on and not add personalized copy to clipboard messages in NVDA for every application. Sighted people do not have such messages either when pressing ctrl+c. So if the application adds such messages itself, then it is a happy side effect for screen reader users.
If we add a copy message for formated copy action in NVDA, then people will wonder why this works in browse mode but not in focus mode, why the message is different between browse and focus mode etc.
At least for raw vs. formated copy actions we can easily explain the difference. and this is something that should be done in the user guide.
